src/main/splitting/task_checker.py

```diff
@@ -90,18 +93,18 @@ def get_no_need_to_run_tests_values(rate: float, tasks_len: int) -> Tuple[bool,
 
     def create_source_file_with_name(self, source_code: str, name: str) -> str:
         source_code_file = os.path.join(TASKS_TESTS_PATH, SOURCE_OBJECT_NAME,
-                                        name + str(get_extension_by_language(self.language).value))
+                                        name + get_extension_by_language(self.language).value)
         create_file(source_code, source_code_file)
         return source_code_file
 
-    def check_before_tests(self, source_file: str, source_code: str, tasks: list):
+    def check_before_tests(self, source_file: str, source_code: str, tasks: List[TASK]) -> (bool, List[float], float):
```

Comment:
I think this function requires description because we have the difficult return type